### PR TITLE
ocamlPackages.duration: 0.2.0 → 0.2.1

### PR DESCRIPTION
**Pull request by vbgl**
###### Description of changes

Minor fix: https://github.com/hannesm/duration/blob/v0.2.1/CHANGES.md

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/ocaml-modules/mirage-time/unix.nix [hunk 1]
```diff
@@ -1,11 +1,12 @@
-{ buildDunePackage, fetchurl, mirage-time, ocaml_lwt, duration }:
+{ buildDunePackage, fetchurl, mirage-time, lwt, duration }:
```
Comment (superherointj) [inline pkgs/development/ocaml-modules/mirage-time/unix.nix:1]:
```suggestion
{ buildDunePackage, fetchurl, mirage-time, lwt, duration }:
```
lwt is at arguments but not added to `propagatedBuildInputs`.

https://github.com/ocaml/opam-repository/blob/master/packages/mirage-time-unix/mirage-time-unix.3.0.0/opam#L24

I get that this package will inherit `lwt` (indirectly) from `mirage-time`. Both ways will obviously work. But is there an ideal way to package this? Should we be listing all dependencies or rely on indirect dependencies?

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 220543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>comby</li>
    <li>dune-release</li>
    <li>jackline</li>
    <li>ligo</li>
    <li>liquidsoap</li>
  </ul>
</details>

File: pkgs/development/ocaml-modules/mirage-time/unix.nix [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
 buildDunePackage {
   pname = "mirage-time-unix";
 
-  inherit (mirage-time) src useDune2 version minimumOCamlVersion;
+  inherit (mirage-time) src version;
+  duneVersion = "3";
 
-  propagatedBuildInputs = [ mirage-time ocaml_lwt duration ];
+  propagatedBuildInputs = [ mirage-time duration ];
```
[the remark this answers is not in the document]
Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-time/unix.nix:9]: Yes, this is what I meant… Let me fix the commit.

[the remark this answers is not in the document]
Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-time/unix.nix]: Done.

File: pkgs/development/ocaml-modules/mirage-time/unix.nix (hunk 1 above)
Comment (vbgl) [reply pkgs/development/ocaml-modules/mirage-time/unix.nix:1]: Well spotted. This should be fixed now. I don’t know what is ideal.

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 220543` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>comby</li>
    <li>dune-release</li>
    <li>jackline</li>
    <li>ligo</li>
    <li>liquidsoap</li>
  </ul>
</details>

**Review by superherointj — APPROVED** [review]
LGTM. Sorry about delay.

**Pull request merged**